frontend/javascript/basket.js

```diff
@@ -0,0 +1,124 @@
+            updateTotal();
+        });
+    });
+}
+
+function removeItemFromCartList(currentCartItemId) {
+    const start = cartList.indexOf(currentCartItemId);
+    const deleteCount = 1;
+    return cartList.splice(start, deleteCount);
+}
+function updateTotal() {
+    let cartBoxes = document.querySelector('.cart-content').querySelectorAll('.cart-box');
+
+    let totalPrice = document.querySelector('.total__price');
+    let total = 0;
+
+    cartBoxes.forEach((cartBox) => {
+        let priceElement = cartBox.querySelector('.cart-box__product_price');
+        let price = parseFloat(priceElement.innerHTML.replace('£', ''));
+        let quantity = cartBox.querySelector('.cart-box__product-quantity').innerHTML;
+        total += price * quantity;
+    });
+
+    total = total.toFixed(2);
+
+    totalPrice.innerHTML = '£' + total;
+}
+
+function initiateQuantityButtons() {
+    let cartBoxes = document
+        .querySelector('.cart-content')
+        .querySelectorAll('.cart-box_quantity-control');
+
+    cartBoxes.forEach((cartBox) => {
+        let decreaseQtyImg = cartBox.querySelector('.cart-box-btn__disabled ');
+        let currentQuantityElement = cartBox.querySelector('.cart-box__product-quantity');
+        let currentQuantity = Number(currentQuantityElement.innerHTML);
+
+        let increaseInQuantitybtn = cartBox.querySelector('.cart-box_quantity-increase');
+        increaseInQuantitybtn.addEventListener('click', function () {
+            decreaseQtyImg.className = 'cart-box-btn__enabled';
+            currentQuantity += 1;
+            currentQuantityElement.innerHTML = currentQuantity;
+            updateTotal();
+        });
+
+        let decreaseInQuantitybtn = cartBox.querySelector('.cart-box_quantity-decrease');
+        decreaseInQuantitybtn.addEventListener('click', function () {
+            const hasMinimumAllowedQuantity = isUpdatedQuantityOne(currentQuantity - 1);
+
+            currentQuantity = decrementQuantity(currentQuantity);
+            currentQuantityElement.innerHTML = currentQuantity;
+            updateTotal();
+
+            decreaseQtyImg.className = hasMinimumAllowedQuantity
+                ? 'cart-box-btn__disabled'
+                : 'cart-box-btn__enabled';
+        });
+    });
+}
+
+// TODO save quantity in session storage and make the decrease button faded out in index HTLM and when quantity increases turn it normal
+
+function isUpdatedQuantityOne(quantity) {
+    return quantity <= 1;
+}
+
+function decrementQuantity(quantity) {
+    return quantity <= 1 ? 1 : quantity - 1;
+}
+
+const supportsTemplate = () => 'content' in document.createElement('template');
+
+let doesBrowserSupportTemplete = supportsTemplate();
+
+function addCartItem(addCartBtn, productList) {
+    if (doesBrowserSupportTemplete) {
+        let templateClone = createTemplate('#cart-box-template');
+        let currentProductId = addCartBtn.getAttribute('data-test');
+
+        if (cartList.includes(currentProductId)) {
```

Comment:
I know this is currently being refactored, but in case the logic is to be reused - can we omit the `else` statement here, and move its body up to the `if` (currently empty) statement? And just invert the condition that's being used for the `if`?

Reply:
If statement logic has been inverted